src/components/root_layout.tsx

```diff
@@ -27,21 +27,26 @@ const RootLayout: FC<RootLayoutProps> = ({ children }) => {
 // back button should send the user to the JobsView. templateName set to the empty string while 
 // showBackButton is true indicates that the back button should send the user to the Home screen.
 const [templateName, setTemplateName] = useState("");
-const [showBackButton, setShowBackButton] = useState(false);
+const [showBackButtonToTemplate, setShowBackButtonToTemplate] = useState(false);
```

Comment:
I don't think you need to change the state variables.  Instead, just change the <Link to=... code.
```
to={templateName ? `app/{templateName}` : '/'}
```

Reply:
I didn't think about this, this is way easier, but it might cause problem if we later have more back button

Reply:
I really like your use of the `backUrl` state variable. That provides any flexibility we may need in the future. Good idea.